packages/core/src/components/graphs/confidence-interval.ts

```diff
@@ -0,0 +1,208 @@
+// Internal Imports
+import { Component } from "../component";
+import * as Configuration from "../../configuration";
+import {
+	CartesianOrientations,
+	Events,
+	ColorClassNameTypes
+} from "../../interfaces";
+import { DOMUtils } from "../../services";
+import { Tools } from "../../tools";
+import settings from "carbon-components/es/globals/js/settings";
+
+// D3 Imports
+import { area } from "d3-shape";
+import { select } from "d3-selection";
+
+export class ConfidenceInterval extends Component {
+	type = "area";
+
+	init() {
+		const eventsFragment = this.services.events;
+
+		// Highlight correct area on legend item hovers
+		eventsFragment.addEventListener(
+			Events.Legend.ITEM_HOVER,
+			this.handleLegendOnHover
+		);
+
+		// Un-highlight area on legend item mouseouts
+		eventsFragment.addEventListener(
+			Events.Legend.ITEM_MOUSEOUT,
+			this.handleLegendMouseOut
+		);
+	}
+
+	render(animate = true) {
+		const svg = this.getContainerSVG({ withinChartClip: true });
+		let domain = [0, 0];
+
+		const { cartesianScales } = this.services;
+
+		const orientation = cartesianScales.getOrientation();
+		const areaGenerator = area().curve(this.services.curves.getD3Curve());
+
+		// Update the bound data on area groups
+		const groupedData = this.model.getGroupedData();
+
+		const confidence = groupedData && groupedData.length === 1;
+
+		if (!confidence) {
+			console.warn(
+				"Confidence Intervals can only be shown when having 1 single dataset"
```

Comment:
```suggestion
				`Confidence Intervals can only be shown when having 1 single datagroup, you've supplied ${groupedData.length}`
```

Reply:
Also, if they provide multiple datagroups, what do they see?

Reply:
The change suggested has been implemented. If multiple datagroups are provided tho you just see the lines without the confidence attached, as pictured below. Is that ok?
<img width="754" alt="Screenshot 2020-11-27 at 15 44 17" src="https://user-images.githubusercontent.com/44590418/100465597-7dac4880-30c7-11eb-9677-4d3c76926fb1.png">

Reply:
I was thinking maybe the 1st one just shows up with the confidence, but I guess this works too